packages/components/src/Chips/InternalChipDismissible/hooks/useInternalChipDismissibleInput.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {
+  ChangeEvent,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { v1 as uuidV1 } from "uuid";
+import { debounce } from "lodash";
+import {
+  ChipDismissibleInputOptionProps,
+  ChipDismissibleInputProps,
+} from "../InternalChipDismissibleTypes";
+import { Icon } from "../../../Icon";
+import { ChipProps } from "../../Chip";
+
+const menuId = uuidV1();
+
+export function useInternalChipDismissibleInput({
```

Comment:
Is the naming for this verbose?  Would something slightly simpler like `getOptionsAndActions` or something like that work? I saw `RevealButton` and a `useReveal` hook in mobile, but that was a short name to begin with, I wonder if this is based on a naming convention I am unaware of.

Reply:
Eh, was trying to name it the same as the parent (+ "use") so no one uses it outside if it. Imagine all components having that simple name. Auto import is gonna be hella confusing.

Reply:
Ya that was a thought I had around importing too (that there might be quite a bit of confusion).  It ain't a big deal either way, just thought the name was getting quite long, that's all :)

Reply:
I'd be concerned about the name if it's a re-useable thing by any component.